test/test.js

```diff
@@ -174,13 +174,55 @@ describe('typeis.hasBody(req)', function () {
       assert.strictEqual(typeis.hasBody(req), true)
     })
   })
+
+  describe('http2 request', function () {
+    it('should not indicate body', function () {
+      var req = {
+        headers: {},
+        stream: {
+          _readableState: {
+            ended: true
+          }
+        },
+        httpVersionMajor: 2
+      }
+      assert.strictEqual(typeis.hasBody(req), false)
+    })
+
+    it('should indicate body', function () {
+      var req = {
+        headers: {},
+        stream: {
+          _readableState: {
+            ended: false
+          }
+        },
+        httpVersionMajor: 2
+      }
+      assert.strictEqual(typeis.hasBody(req), true)
+    })
+  })
 })
 
 function createRequest (type) {
-  return {
-    headers: {
-      'content-type': type || '',
-      'transfer-encoding': 'chunked'
+  if (process.env.HTTP2_TEST) {
+    return {
+      headers: {
+        'content-type': type || ''
+      },
+      stream: {
+        _readableState: {
```

Comment:
At this point if we are depending on Node.js internals for the state, we need to test against the actual Node.js implementation instead of a simple mock.

Reply:
Either that or there is specific documentation in Node.js documenting this exact behavior, though I didn't see that anywhere. Without one of these we won't know when Node.js changes it's behavior as our tests won't fail.

Reply:
I will fix tests tomorrow.